File: Ex18Fragment/app/src/main/java/com/hsr2024/ex18fragment/Myfragment.java
package com.hsr2024.ex18fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.w3c.dom.Text;

public class Myfragment extends Fragment {

    TextView tv;
    Button btn,btn2;

    //프래그먼트가 화면에 보여줘야할 View 객체를 생성하여 리턴해주는 콜백메소드 - 라이프사이클
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        //layout폴더 안에 있는 fragment_my.xml 문서를 읽어와서 뷰 객체로 만들어주는 작업..
        View view= inflater.inflate(R.layout.fragment_my, container, false);
        return view;
    }

    //프래그먼트가 보여줄 뷰객체를 만들어 낸 후.. 리스너 연결 작업 등을 수행하는 작업영역 메소드


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        tv= view.findViewById(R.id.tv);
        btn= view.findViewById(R.id.btn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                tv.setText("Nice to meet you...");
            }
        });

        btn2= view.findViewById(R.id.btn2);
        btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //MainActivity에 있는 TextView를 제어하기
                //먼저 이 프레그먼트를 보여주는 액티비티 객체를 참조하는 기능메소드
                MainActivity ma= (MainActivity) getActivity();
                ma.tv.setText("으헤헤헤헤헤...");
            }
        });
    }
}

File: Test60/app/src/main/java/com/hsr2024/test60/MyFragment.kt
package com.hsr2024.test60

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.hsr2024.test60.databinding.FragmentMyBinding

class MyFragment:Fragment() {

    lateinit var binding: FragmentMyBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding= FragmentMyBinding.inflate(inflater,container,false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.btn.setOnClickListener { binding.tv.text = "방가방가~" }
        binding.btn2.setOnClickListener {
            val mainActivity:MainActivity = activity as MainActivity
            mainActivity.binding.tv.text = " 프래그먼트 버튼으로 메인 조종하기"
        }


    }
}